eng/common/mcp/README.md

```diff
@@ -24,15 +24,13 @@ The script will install the latest version of the azsdk cli executable from [too
 
 Azure SDK MCP server code is in [azure-sdk-tools/tools/azsdk-cli/Azure.Sdk.Tools.Cli](https://github.com/Azure/azure-sdk-tools/tree/main/tools/azsdk-cli/Azure.Sdk.Tools.Cli).
 
-Azure SDK MCP servers should support [stdio and sse transports](https://modelcontextprotocol.io/docs/concepts/transports#server-sent-events-sse).
+Azure SDK MCP servers should support [stdio and streamable HTTP transports](https://modelcontextprotocol.io/docs/concepts/transports).
```

Comment:
MUST support stdio and SHOULD support StreamableHttp using latest OAuth 2.1 Best Current Practices (BCPs)

Reply:
@joshfree do you mean to say if we are adding http it must require auth?

I originally had this sse/http line in here to ensure that servers would be debuggable via inspector, IDE breakpoints, etc.